d3a_interface/data_classes.py

```diff
@@ -197,6 +215,24 @@ def serializable_dict(self) -> Dict:
                 "buyer": self.buyer,
                 }
 
+    @staticmethod
+    def from_dict(bid: Dict) -> "Bid":
+        """Deserialize a bid dict."""
+
+        return Bid(
+            id=bid["id"],
+            time=str_to_pendulum_datetime(bid["time"]),
+            energy=bid["energy"],
+            price=bid["energy"] * bid["energy_rate"],
+            original_price=bid.get("original_price"),
+            buyer=bid.get("buyer"),
+            buyer_origin=bid.get("buyer_origin"),
+            buyer_origin_id=bid.get("buyer_origin_id"),
```

Comment:
```suggestion
            buyer_origin=bid["buyer_origin"],
            buyer_origin_id=bid["buyer_origin_id"],
```
Can you be a bit strict here, to guarantee its there and not being overlooked.

Reply:
@faizan2590 I'm afraid this is not possible at the moment, this will require updating all references (mostly in tests) to pass this variable, which is out of the context of this PR imo